src/extension.ts

```diff
@@ -279,3 +282,9 @@ async function summarizeResultsAsToast(targets: string[], results: RuleResult[])
 export function deactivate() {
 	TelemetryService.dispose();
 }
+
+export function _isValidFileForAnalysis(documentUri: vscode.Uri) {
+	const allowedFileTypes:string[] = ['.cls', '.js', '.apex', '.trigger', '.ts'];
+	return allowedFileTypes.includes(path.extname(documentUri.path));
```

Comment:
Do we care about case insensitivity when validating file extensions?

Reply:
If someone creates apex class or any of these files using the extension, I'm pretty these are going to be in lower case. I have seen similar code in vscode extension where the checks are made only for lower case file. I think it is good as is.

Reply:
Fair enough. Approved.